src/world/Actor/Chara.cpp

```diff
@@ -551,28 +517,44 @@ void Sapphire::Entity::Chara::addStatusEffect( StatusEffect::StatusEffectPtr pEf
   status.index = static_cast< uint8_t >( nextSlot );
   status.param = pEffect->getParam();
 
-  sendToInRangeSet( statusEffectAdd, isPlayer() );
+  float totalShieldValue = 0;
+  for( auto effectIt : m_statusEffectMap )
+  {
+    auto statusEffect = effectIt.second;
+    if( static_cast< Common::StatusEffectType >( statusEffect->getEffectEntry().effectType ) == Common::StatusEffectType::Shield )
+    {
+      totalShieldValue += statusEffect->getEffectEntry().effectValue1;
+    }
+  }
+
+  if( totalShieldValue > 0 )
+  {
+    totalShieldValue /= getMaxHp();
+    totalShieldValue *= 100;
+    statusEffectAdd->data().shieldPercentage = totalShieldValue >= 255 ? 255 : static_cast< uint8_t >( totalShieldValue );
+  }
```

Comment:
why is all this code duplicated again later in chara?

https://github.com/SapphireServer/Sapphire/pull/626/files#diff-0a3d3fe0bb35313cf639920ab3545a7cR689

Reply:
also use std::min instead of this ternary crap

Reply:
two different packets have this same shield field need to be sent, and not worth it to create a function for this since one of them had an existing loop can be used instead of looping it again just for this, and only clients care about this percentage anyway